### PR TITLE
chore: init zk-kit.circom repository

### PR DESCRIPTION
**Pull request by cedoor**
This PR initializes a new repo for hosting Circom circuits, copying the circuits from the [ZK-Kit](https://github.com/privacy-scaling-explorations/zk-kit) repository as part of a new structure of the ZK-Kit packages. ZK-Kit will have a repository per language.

Re https://github.com/privacy-scaling-explorations/zk-kit/issues/282

### CODE REVIEW

**Comment by 0xjei** [comment]
the `utils` package circuit aren't building from `compile` root script. Got the following error:

```bash
[@zk-kit/utils.circom]: Process started
[@zk-kit/utils.circom]: ===| Compiling the circuit |===
[@zk-kit/utils.circom]: Error: No such circuit in ./circuits.json
```

**Comment by 0xjei** [comment]
> > the `utils` package circuit aren't building from `compile` root script. Got the following error:
> 
> I should have solved it! Let me know 👍🏽

yes, works fine - thank you for the quick fix

File: packages/poseidon-proof/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+    "name": "@zk-kit/poseidon-proof.circom",
+    "version": "1.0.0-beta",
+    "description": "A circuit to prove the possession of Poseidon pre-images without revealing the pre-images themselves.",
```
Comment (0xjei) [inline packages/poseidon-proof/package.json:4]:
```suggestion
    "description": "A circuit to prove the ownership of Poseidon pre-images without revealing them.",
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Contributing
+
+:tada: Thank you for being interested in contributing to the ZK-kit project! :tada:
```
Comment (0xjei) [inline CONTRIBUTING.md:3]:
```suggestion
:tada: Thank you for being interested in contributing to the ZK-kit Circom project! :tada:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+Feel welcome and read the following sections in order to know how to ask questions and how to work on something.
+
+All members of our community are expected to follow our [Code of Conduct](/CODE_OF_CONDUCT.md). Please make sure you are welcoming and friendly in all of our spaces.
+
+We're really glad you're reading this, because we need volunteer developers to help this project come to fruition. 👏
+
+## Issues
+
+The best way to contribute to our projects is by opening a [new issue](https://github.com/privacy-scaling-explorations/zk-kit/issues/new/choose) or tackling one of the issues listed [here](https://github.com/privacy-scaling-explorations/zk-kit/contribute).
```
Comment (0xjei) [inline CONTRIBUTING.md:13]:
```suggestion
The best way to contribute to our projects is by opening a [new issue](https://github.com/privacy-scaling-explorations/zk-kit.circom/issues/new/choose) or tackling one of the issues listed [here](https://github.com/privacy-scaling-explorations/zk-kit.circom/contribute).
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+            🔎 Issues
+        </a>
+        <span>&nbsp;&nbsp;|&nbsp;&nbsp;</span>
+        <a href="https://appliedzkp.org/discord">
+            🗣️ Chat &amp; Support
+        </a>
+    </h4>
+</div>
+
+| ZK-Kit is a set of libraries (plugins, algorithms or utility functions) that can be reused in different projects and zero-knowledge protocols, making it easier for developers to access user-friendly, tested, and documented code for common tasks. ZK-Kit provides a repository for each language. This repository only contains Circom code. |
```
Comment (0xjei) [inline README.md:46]:
```suggestion
| ZK-Kit is a set of libraries (plugins, algorithms or utility functions) that can be reused in different projects and zero-knowledge protocols, making it easier for developers to access user-friendly, tested, and documented code for common tasks. ZK-Kit provides a repository for each language. This repository only contains Circom templates. |
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+
+> [!IMPORTANT]  
+> Installation of [Circom](https://docs.circom.io/getting-started/installation/) required for circuit tests.
```
Comment (0xjei) [inline README.md:50]: +1

File: package.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "zk-kit",
```
Comment (0xjei) [inline package.json:2]:
```suggestion
    "name": "zk-kit.circom",
```

**Review by 0xjei — COMMENTED** [review]
amazing, makes sense to have one package per circuit - the configs are lit, just a few nits!

**Comment by cedoor** [comment]
> amazing, makes sense to have one package per circuit - the configs are lit, just a few nits!

Thank you so much! Applied all changes you suggested 👍🏽

**Pull request merged**